src/main/java/it/gov/pagopa/pu/registry/mapper/pagopa/RegistryEventPagoPaDTO2PagoPaRegistryMapper.java

```diff
@@ -0,0 +1,80 @@
+package it.gov.pagopa.pu.registry.mapper.pagopa;
+
+import it.gov.pagopa.pu.registry.dto.RegistryEventPagoPaDTO;
+import it.gov.pagopa.pu.registry.enums.RegistryEventCategory;
+import it.gov.pagopa.pu.registry.enums.RegistryEventSubType;
+import it.gov.pagopa.pu.registry.enums.RegistryOutcome;
+import it.gov.pagopa.pu.registry.enums.RegistryPagopaEventType;
+import it.gov.pagopa.pu.registry.model.PagopaRegistry;
+import it.gov.pagopa.pu.registry.service.DataCipherService;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.IntStream;
+
+@Service
+public class RegistryEventPagoPaDTO2PagoPaRegistryMapper {
+  private final DataCipherService dataCipherService;
+
+  public RegistryEventPagoPaDTO2PagoPaRegistryMapper(DataCipherService dataCipherService) {
+    this.dataCipherService = dataCipherService;
+  }
+
+  public List<PagopaRegistry> mapToPagoPaRegistry(RegistryEventPagoPaDTO dto) {
+    List<PagopaRegistry> result = new ArrayList<>();
+
+    String iuvRaw = dto.getIuv();
+    String navRaw = dto.getNav();
+
+    boolean iuvEmpty = (iuvRaw == null || iuvRaw.isBlank());
+    boolean navEmpty = (navRaw == null || navRaw.isBlank());
+
+    if (iuvEmpty && navEmpty) {
+      PagopaRegistry pagopaRegistry = buildRegistry(dto, null, null);
+      result.add(pagopaRegistry);
+    } else {
+      List<String> iuvList = iuvEmpty ? Collections.emptyList() :
+        Arrays.stream(iuvRaw.split(",")).map(String::trim).toList();
+
+      List<String> navList = navEmpty ? Collections.emptyList() :
+        Arrays.stream(navRaw.split(",")).map(String::trim).toList();
+
+      if (iuvList.size() == navList.size()) {
+        IntStream.range(0, iuvList.size())
+          .mapToObj(i -> buildRegistry(dto, iuvList.get(i), navList.get(i)))
+          .forEach(result::add);
+      } else {
+        throw new IllegalArgumentException("mapToPagoPaRegistry - The size of the IUV list (" +
+          iuvList.size() + ") does not match the size of the NAV list (" + navList.size() + ").");
+      }
```

Comment:
this logic is dangerous...
we could lost data this way, because you will ignore all data if it occurs...
Moreover, this exception is not catched and handled in other ways...
We could incur into "hot potato" anti-pattern (this is not true because SpringCloudStream will limit to 3 retry before to ack the message and continue with following messages)

let's instead publish each message, letting null if one of the 2 lists is ended before